### PR TITLE
HPCC-14246 Allow DynamicESDL service methods to be written in JAVA

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck anthony.fishbeck@lexisnexis.com

### CODE REVIEW

File: plugins/javaembed/javaembed.cpp [hunk 1]
```diff
@@ -1153,6 +1157,272 @@ class JavaRowStream : public CInterfaceOf<IRowStream>
...
+        jobject fieldObj = (jobject) JNIenv->GetObjectField(parentObject, fieldId);
+        if (!fieldObj)
+            return;
+        writer.outputBeginNested(fieldname, true);
+        writeChildren(JNIenv->GetObjectClass(fieldObj), fieldObj, defStruct);
+        writer.outputEndNested(fieldname);
+    }
+    void writeSimpleArray(jclass parentClass, jobject parentObject, IEsdlDefObject &defObject)
+    {
+        //tbd
```
Comment (richardkchapman) [inline plugins/javaembed/javaembed.cpp]: Is there a Jira open to make sure you revisit this? Should it say UNIMPLEMENTED; ?

File: plugins/javaembed/javaembed.cpp [hunk 2]
```diff
@@ -1694,14 +1994,404 @@ class JavaThreadContext
...
+    {
+        javaembed::checkException(JNIenv, false);
+    }
+
+    void initWriter()
+    {
+    }
+    IXmlWriterExt & clear()
+    {
+        UNIMPLEMENTED;
```
Comment (richardkchapman) [inline plugins/javaembed/javaembed.cpp]: Does this need to be implemented sometime, or should it be a throwUnexpected(); ?

**Comment by richardkchapman** [comment]
@rpastrana Please review

File: plugins/javaembed/javaembed.cpp (hunk 2 above)
Comment (afishbeck) [inline plugins/javaembed/javaembed.cpp]: right, should be throwUnexpected() for now at least.

File: plugins/javaembed/javaembed.cpp (hunk 1 above)
Comment (afishbeck) [inline plugins/javaembed/javaembed.cpp]: it's an internal placeholder that I don't even call yet.  I'll open a JIRA.

**Comment by afishbeck** [comment]
@richardkchapman @rpastrana updated based on comments

**Comment by afishbeck** [comment]
Also created https://track.hpccsystems.com/browse/HPCC-14256 subtask regarding handling ESDL simple arrays.

**Comment by rpastrana** [comment]
Latest commit seems to address all code review issues. @afishbeck @richardkchapman

**Pull request merged**